frontend/rust-lib/flowy-grid/src/event_map.rs

```diff
@@ -42,63 +42,136 @@ pub fn create(grid_manager: Arc<GridManager>) -> Module {
+/// out, it includes how to use these annotations: input, output, etc.
 #[derive(Clone, Copy, PartialEq, Eq, Debug, Display, Hash, ProtoBuf_Enum, Flowy_Event)]
 #[event_err = "FlowyError"]
 pub enum GridEvent {
+    /// [GetGrid] event is used to get the [GridPB]
+    ///
+    /// The event handler accepts [GridIdPB] and return [GridPB] if there is no errors.
+    ///
     #[event(input = "GridIdPB", output = "GridPB")]
     GetGrid = 0,
 
+    /// [GetGridBlocks] event is used to get the grid's block.
+    ///
+    /// The event handler accepts [QueryGridBlocksPayloadPB] and return [RepeatedGridBlockPB]
+    /// if there is no errors.
     #[event(input = "QueryGridBlocksPayloadPB", output = "RepeatedGridBlockPB")]
     GetGridBlocks = 1,
 
+    /// [GetGridSetting] event is used to get the grid's setting.
+    ///
+    /// The event handler accepts [GridIdPB] and return [GridSettingPB]
+    /// if there is no errors.
     #[event(input = "GridIdPB", output = "GridSettingPB")]
     GetGridSetting = 2,
 
+    /// [UpdateGridSetting] event is used to update the grid's setting.
+    ///
+    /// The event handler accepts [GridIdPB] and return errors if failed to modify the grid's setting.
     #[event(input = "GridIdPB", input = "GridSettingChangesetPayloadPB")]
     UpdateGridSetting = 3,
 
+    /// [GetFields] event is used to get the grid's setting.
+    ///
+    /// The event handler accepts [QueryFieldPayloadPB] and return [RepeatedGridFieldPB]
+    /// if there is no errors.
     #[event(input = "QueryFieldPayloadPB", output = "RepeatedGridFieldPB")]
     GetFields = 10,
 
+    /// [UpdateField] event is used to update the field attributes.
+    ///
+    /// The event handler accepts [FieldChangesetPayloadPB] and return errors if failed to modify the
+    /// field.
     #[event(input = "FieldChangesetPayloadPB")]
     UpdateField = 11,
 
+
+    /// [UpdateFieldTypeOption] event is used to update the field's type option data. Certain field
+    /// types have user-defined options such as color, date format, number format, or a list of values
+    /// for a multi-select list. These options are defined within a specialization of the
+    /// FieldTypeOption class.
+    ///
+    /// Check out [this](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/architecture/frontend/grid#fieldtype)
+    /// for more information.
+    ///
+    /// The event handler accepts [UpdateFieldTypeOptionPayloadPB] and return errors if failed to modify the
+    /// field.
     #[event(input = "UpdateFieldTypeOptionPayloadPB")]
     UpdateFieldTypeOption = 12,
 
+    /// [InsertField] event is used to insert a new field. If the field is already exists, the event
+    /// handler will replace the value with the new field value.
     #[event(input = "InsertFieldPayloadPB")]
     InsertField = 13,
 
+    /// [DeleteField] event is used to delete a new field. [DeleteFieldPayloadPB] is the context that
```

Comment:
to delete a "new" field? I think you mean "to delete a Field" (I already changed this in my commit. Just ignore if removing "new" is ok)